lib/Sema/CSSimplify.cpp

```diff
@@ -10647,8 +10647,108 @@ ConstraintSystem::simplifyApplicableFnConstraint(
         subKind, argumentsLoc, trailingClosureMatching);
 
     switch (matchCallResult) {
-    case SolutionKind::Error:
+    case SolutionKind::Error: {
+      if (shouldAttemptFixes())
+        return SolutionKind::Error;
```

Comment:
Hmm, does this mean we won't attempt to re-match the trailing closure to `callAsFunction` in diagnostic mode? Will that lead to any misleading diagnostics if there are other errors in the closure or something like that?

Reply:
Looks like this is changed in https://github.com/apple/swift/pull/41189/commits/1c258ca05f37b7081765df742262887291c41fc6

Reply:
Yeah, sorry I was testing changes and forgot to remove it initially and indeed injection should be attempted in both modes.